modules/hostap/src/supp_api.c

```diff
@@ -403,6 +404,25 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 		}
 	}
 
+	memcpy((void *)&mac, params->bssid, WIFI_MAC_ADDR_LEN);
+	if (net_eth_is_addr_broadcast(&mac) ||
```

Comment:
You move this check in to the below branch (!unspecified)

Reply:
!unspecified checks for all 0s which is equivalent to `not setting the BSSID`. That is why I am returning error only for broadcast and multicast case.